src/model.ts

```diff
@@ -971,6 +979,17 @@ export class SpraypaintBase {
     })
   }
 
+  private async _handleAcceptedResponse(response: any): Promise<any> {
+    let returnValue = await this._handleResponse(response, () => {})
```

Comment:
For your use case, are you ever actually using this response? I'm not wild about always returning a boolean except for this very specific case - I'd prefer we return a boolean, but fire a method like `onDeferredDestroy` or something similar.

Reply:
are you thinking about an optional argument into save/delete to pass the `onDeferredDestroy`

Reply:
@richmolj

Reply:
I'm thinking that we always return `true`, but before that call an `onDeferredDestroy()` method. We can define that method in `SpraypaintBase`, doing nothing by default.

Reply:
So it's up to the specific Model class to decide/implement the method `onDeferredDestroy()` ? And will that implementation call 
```
      this.klass.fromJsonapi(
        response.jsonPayload.data,
        response.jsonPayload
      )
```

I'm thinking of this:

```js
class SpraypaintBase {
  onDeferredDestroy(response: any): void {
  }

  async destroy(): Promise<any> {
    const url = this.klass.url(this.id)
    const verb = "delete"
    const request = new Request(this._middleware(), this.klass.logger)
    let response: any

    try {
      response = await request.delete(url, this._fetchOptions())
    } catch (err) {
      throw err
    }

    if (response.status === 202) {
      return await this._handleAcceptedResponse(response)
    }

    let base = this.klass.baseClass as typeof SpraypaintBase
    base.store.destroy(this)

    return await this._handleResponse(response, () => {
      this.isPersisted = false
    })
  }

  private async _handleAcceptedResponse(response: any): Promise<any> {
    this.onDeferredDestroy(response)
    return await this._handleResponse(response, () => {})
  }
}

class StuffBatch extends SpraypaintBase {
  onDeferredDestroy(response: any): void {
    const returnedObject = this.klass.fromJsonapi(
      response.jsonPayload.data,
      response.jsonPayload
    )
    // do things with returnedObject
  }
}
```
What do you think?

Reply:
Yep! A little hard to tell without a diff but I think you've got it! Does that work for you?

Reply:
I can give it a shot and see how it goes.